server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1708,7 +1708,8 @@ static void prepareResizeIndexSettings(
             .put(builder.build())
             .put(IndexMetadata.SETTING_ROUTING_PARTITION_SIZE, sourceMetadata.getRoutingPartitionSize())
             .put(IndexMetadata.INDEX_RESIZE_SOURCE_NAME.getKey(), resizeSourceIndex.getName())
-            .put(IndexMetadata.INDEX_RESIZE_SOURCE_UUID.getKey(), resizeSourceIndex.getUUID());
+            .put(IndexMetadata.INDEX_RESIZE_SOURCE_UUID.getKey(), resizeSourceIndex.getUUID())
+            .put(INDEX_REPLICATION_TYPE_SETTING.getKey(), INDEX_REPLICATION_TYPE_SETTING.get(sourceMetadata.getSettings()));
```

Comment:
If the replication type cannot be changed, how about we throw an error when users set a different value for that setting when shrink/clone/split? Similar to the behavior:https://github.com/opensearch-project/OpenSearch/blob/ae22e3ff32ef15a6af302c50872f1fa0e8e140fe/server/src/main/java/org/opensearch/action/admin/indices/shrink/TransportResizeAction.java#L312.

If we allow configuring a different setting value but doesn't adopt it, it is a little bit weird and introduces confusion.

Reply:
LGTM

Reply:
FIXED At https://github.com/opensearch-project/OpenSearch/pull/16076/commits/81aec5e113bd19a9228133ecd671a7a021421eba